packages/preferences/src/common/preference-service.ts

```diff
@@ -32,60 +34,70 @@ export class PreferenceService implements Disposable {
         this.toDispose.dispose();
     }
 
+    ready(): Promise<void> {
+        return this.isReady;
+    }
+
     protected onDidChangePreference(event: PreferenceChangedEvent): void {
+        for (const prefChange of event.changes) {
+            // Pref removed
+            if (prefChange.oldValue !== undefined && prefChange.newValue === undefined) {
+                delete this.prefCache[prefChange.preferenceName];
+            } else if (prefChange.newValue !== undefined) {
+                this.prefCache[prefChange.preferenceName] = prefChange.newValue;
+            }
+        }
+
+        Promise.resolve(this.ready);
```

Comment:
no, `Promise.resolve(this.ready)`, just returns another promise that is waiting when `this.ready` is resolved, it does not resolve it